server/main_server/src/main.cpp

```diff
@@ -1567,6 +1547,128 @@ Environment Variables:
         };
     } // remove_leftover_agent_info_files_for_ips
 
+    auto init_access_time_queue() -> std::optional<std::string>
+    {
+        try {
+            //
+            // Retrieve access time configuration from the catalog.
+            //
+
+            std::optional<std::string> queue_name_prefix;
+            std::optional<std::string> queue_size;
+            std::optional<std::string> resolution_in_seconds;
+
+            const auto role = irods::get_server_property<std::string>(irods::KW_CFG_CATALOG_SERVICE_ROLE);
+
+            // Initialization of the access time queue happens before the agent factory is available.
+            // This means we cannot rely on a local iRODS connection to retrieve the access time configuration 
+            // from the database.
+
+            if (role == irods::KW_CFG_SERVICE_ROLE_PROVIDER) {
+                // Catalog Service Providers are required to have database credentials, so all we have to do
+                // to get around the lack of an agent factory is use nanodbc to interact with the database
+                // directly.
+
+                auto [db_instance, db_conn] = irods::experimental::catalog::new_database_connection();
+                auto row = execute(db_conn, "select option_name, option_value from R_GRID_CONFIGURATION where namespace = 'access_time'");
+
+                while (row.next()) {
+                    const auto opt_name = row.get<std::string>(0);
+
+                    if (opt_name == irods::KW_CFG_ACCESS_TIME_QUEUE_NAME_PREFIX) {
+                        queue_name_prefix = row.get<std::string>(1);
+                    }
+                    else if (opt_name == irods::KW_CFG_ACCESS_TIME_QUEUE_SIZE) {
+                        queue_size = row.get<std::string>(1);
+                    }
+                    else if (opt_name == irods::KW_CFG_ACCESS_TIME_RESOLUTION_IN_SECONDS) {
+                        resolution_in_seconds = row.get<std::string>(1);
+                    }
```

Comment:
what happens if all three are not found in the catalog?

Reply:
found it later... it logs and ... halts startup?   or do we / should we fire it up with hard-coded defaults?

Reply:
Right now, the server just terminates. My initial thought is we shouldn't provide a fallback because that may produce results the admin didn't want.

Reply:
that's fair.   good error message and a non-starting server.    fewer bad states that way.

resolving.